vllm/model_executor/layers/rotary_embedding.py

```diff
@@ -922,11 +922,9 @@ def get_rope(
     else:
         scaling_type = rope_scaling[
             "type"] if "type" in rope_scaling else rope_scaling["rope_type"]
-        # The correct one should be "longrope" but keep "su" here
-        # for backward compatible
-        if scaling_type not in {"su", "longrope"}:
-            scaling_factor = rope_scaling.get("factor", 1.0)
```

Comment:
It is not clear which RoPE implementations need this, so I've moved this code into the individual if blocks.